test/solc/CommandLineParser.cpp

```diff
@@ -0,0 +1,587 @@
+	_out << "    outputDir: " << _options.outputDir << endl;
+	_out << "    overwriteFiles: " << _options.overwriteFiles << endl;
+	_out << "    evmVersion: " << _options.evmVersion.name() << endl;
+	_out << "    experimentalViaIR: " << _options.experimentalViaIR << endl;
+	_out << "    revertStrings: " << _options.revertStrings << endl;
+	_out << "    stopAfter: " << _options.stopAfter << endl;
+	_out << "    inputMode: " << _options.inputMode << endl;
+	_out << "    targetMachine: " << _options.targetMachine << endl;
+	_out << "    inputAssemblyLanguage: " << _options.inputAssemblyLanguage << endl;
+	_out << "    libraries[]: {" << joinHumanReadable(_options.libraries | ranges::views::transform(libraryNameAndAddressToString)) << "}" << endl;
+	_out << "    prettyJson: " << _options.prettyJson << endl;
+	_out << "    coloredOutput: " << _options.coloredOutput << endl;
+	_out << "    withErrorIds: " << _options.withErrorIds << endl;
+	_out << "    selectedOutputs: [" << _options.selectedOutputs << "]" << endl;
+	_out << "    estimateGas: " << _options.estimateGas << endl;
+	if (_options.combinedJsonRequests.has_value())
+		_out << "    combinedJsonRequests: [" << _options.combinedJsonRequests.value() << "]" << endl;
+	else
+		_out << "    combinedJsonRequests: null" << endl;
+	_out << "    metadataHash: " << _options.metadataHash << endl;
+	_out << "    metadataLiteral: " << _options.metadataLiteral << endl;
+	_out << "    optimize: " << _options.optimize << endl;
+	_out << "    expectedExecutionsPerDeployment: " << _options.expectedExecutionsPerDeployment << endl;
+	_out << "    noOptimizeYul: " << _options.noOptimizeYul << endl;
+	if (_options.yulOptimiserSteps.has_value())
+		_out << "    yulOptimiserSteps: " << _options.yulOptimiserSteps.value() << endl;
+	else
+		_out << "    yulOptimiserSteps: null" << endl;
+	_out << "    initializeModelChecker: " << _options.initializeModelChecker << endl;
+	_out << "    modelCheckerSettings: " << _options.modelCheckerSettings << endl;
+	_out << "}";
+
+	if (!originalBoolalpha)
+		_out.unsetf(ios::boolalpha);
+
+	return _out;
+}
+
+} // namespace
+
+namespace boost::test_tools::tt_detail
+{
+
+// Boost won't find find the << operator unless we put it in the std namespace which is illegal.
+// The recommended solution is to overload print_log_value<> struct and make it use our operator.
+
+template<>
+struct print_log_value<CommandLineOptions>
+{
+	void operator()(std::ostream& _out, CommandLineOptions const& _options) { ::operator<<(_out, _options); }
+};
+
+} // namespace boost::test_tools::tt_detail
+
+namespace solidity::frontend::test
+{
+
+BOOST_AUTO_TEST_SUITE(CommandLineParserTest)
+
+BOOST_AUTO_TEST_CASE(no_options)
+{
+	vector<string> commandLine = {"solc", "contract.sol"};
+
+	CommandLineOptions expectedOptions;
+	expectedOptions.sourceFilePaths = {"contract.sol"};
+	expectedOptions.coloredOutput = true;
+	expectedOptions.expectedExecutionsPerDeployment = 200;
```

Comment:
This is a magic number that should appear exactly once in the whole project source.

Reply:
It does appear once in non-test code and I think that's the part that matters. But I'll make it a constant anyway. Less duplication is always good.

Reply:
I mean the question is what do we want to test? Do we want to test that it is the number 200 or do we want to test that the parser sets it to `OptimiserSettings{}.expectedExecutionsPerDeployment`?

Reply:
I'm often OK with hard-coding some things in tests because these values get cross-checked so you have to make a mistake in both places to miss the problem. Deduplicating stuff too much sometimes makes it easy for the wrong value to be carried over to the check in the test so I often prefer to see the "raw" values in tests.

That's my general approach but I get your point about this constant. Expecting a particular raw value is not the point of the test so we should have a constant. The problem is - this value is hard-coded all over the code base. I think this PR is not the best place to fix that. I have created a separate one: #11550.